sheepdog/transactions/upload/entity.py

```diff
@@ -835,6 +855,41 @@ def set_association_proxies(self):
                         msg, keys=[name], type=EntityErrors.INVALID_LINK)
                     continue
 
+                # TT-273, check if we're doing sample->sample for restrictions
+                # TODO: This should be encoded in the dictionary somehow, not here
+                parent_sample_types = [
```

Comment:
Should finish this TODO

Reply:
Agreed, but next dictionary change is end of March at earliest.